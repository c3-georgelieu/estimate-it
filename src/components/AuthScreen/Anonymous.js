import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import {
  Button,
  InputGroup,
  InputGroupAddon,
  Input,
  Alert,
} from 'reactstrap'
import FA from 'react-fontawesome'
import * as api from '../../helpers/api'

const Anonymous = ({
  match: { url, params },
  checkCreds,
  openGuestSession,
}) => (
  <Fragment>
    {(url.startsWith('/estimate/') && params.estimateId !== 'new') ? (
      <Alert color="warning">
        Please enter a valid credentials to get access to this estimate
      </Alert>
    ) : null}
    <form
      onSubmit={e => {
        e.preventDefault()
        const [dbName, apiKey] = e.target.credentials.value.split(':')
        api.setCreds({ dbName, apiKey, username: e.target.username.value })
        checkCreds()
      }}
    >
      <label htmlFor="username">
        Public Name
      </label>
      <InputGroup>
        <InputGroupAddon addonType="prepend">
          <span className="input-group-text">
            <FA name="user" />
          </span>
        </InputGroupAddon>
        <Input
          name="username"
          id="username"
          required
          autoFocus // eslint-disable-line jsx-a11y/no-autofocus
        />
      </InputGroup>
      <label htmlFor="credentials">
        Access Key
      </label>
      <InputGroup>
        <InputGroupAddon addonType="prepend">
          <span className="input-group-text">
            <FA name="key" />
          </span>
        </InputGroupAddon>
        <Input
          type="password"
          name="credentials"
          id="credentials"
          required
        />
      </InputGroup>
      <Button color="primary" outline>
        Sign In
      </Button>
    </form>
    Or
    <br />
    <Button
      color="primary"
      outline
      onClick={openGuestSession}
    >
      Guest Session
    </Button>
  </Fragment>
)

Anonymous.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string.isRequired,
    params: PropTypes.shape({
      estimateId: PropTypes.string,
    }).isRequired,
  }).isRequired,
  checkCreds: PropTypes.func.isRequired,
  openGuestSession: PropTypes.func.isRequired,
}

export default Anonymous
