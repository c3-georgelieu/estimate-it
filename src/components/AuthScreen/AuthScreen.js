import React from 'react'
import PropTypes from 'prop-types'
import {
  Button,
  InputGroup,
  InputGroupAddon,
  Input,
  Alert,
} from 'reactstrap'
import FA from 'react-fontawesome'
import * as api from '../../helpers/api'
import Header from '../Header'

const AuthScreen = ({
  match: { url, params },
  username,
  checkCreds,
  resetCreds,
  cleanEstimate,
  closeAuthScreen,
  openGuestSession,
}) => (
  <div className="overlay auth-screen">
    <Header />
    {username ? (
      <div className="panel">
        <div>
          {/* eslint-disable-next-line react/jsx-one-expression-per-line */}
          Welcome, {username}!
        </div>
        <Button
          color="primary"
          outline
          onClick={closeAuthScreen}
        >
          Close
        </Button>
        <Button
          color="danger"
          outline
          onClick={() => {
            api.removeCreds()
            resetCreds()
            cleanEstimate(params)
          }}
        >
          Log Out
        </Button>
      </div>
    ) : (
      <div className="panel">
        {(url.startsWith('/estimate/') && params.estimateId !== 'new') ? (
          <Alert color="warning">
            Please enter a valid credentials to get access to this estimate
          </Alert>
        ) : null}
        <form
          onSubmit={e => {
            e.preventDefault()
            const [dbName, apiKey] = e.target.credentials.value.split(':')
            api.setCreds({ dbName, apiKey, username: e.target.username.value })
            checkCreds()
          }}
        >
          <label htmlFor="username">
            Public Name
          </label>
          <InputGroup>
            <InputGroupAddon addonType="prepend">
              <span className="input-group-text">
                <FA name="user" />
              </span>
            </InputGroupAddon>
            <Input
              name="username"
              id="username"
              required
              autoFocus // eslint-disable-line jsx-a11y/no-autofocus
            />
          </InputGroup>
          <label htmlFor="credentials">
            Access Key
          </label>
          <InputGroup>
            <InputGroupAddon addonType="prepend">
              <span className="input-group-text">
                <FA name="key" />
              </span>
            </InputGroupAddon>
            <Input
              type="password"
              name="credentials"
              id="credentials"
              required
            />
          </InputGroup>
          <Button color="primary" outline>
            Sign In
          </Button>
        </form>
        Or
        <br />
        <Button
          color="primary"
          outline
          onClick={openGuestSession}
        >
          Guest Session
        </Button>
      </div>
    )}
  </div>
)

AuthScreen.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string.isRequired,
    params: PropTypes.shape({
      estimateId: PropTypes.string,
    }).isRequired,
  }).isRequired,
  username: PropTypes.string.isRequired,
  checkCreds: PropTypes.func.isRequired,
  resetCreds: PropTypes.func.isRequired,
  cleanEstimate: PropTypes.func.isRequired,
  closeAuthScreen: PropTypes.func.isRequired,
  openGuestSession: PropTypes.func.isRequired,
}

export default AuthScreen
