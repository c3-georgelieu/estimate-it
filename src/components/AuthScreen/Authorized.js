import React, { Fragment } from 'react'
import { Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import { Button } from 'reactstrap'
import * as api from '../../helpers/api'
import { locationType } from '../../helpers/propTypes'

const Authorized = ({
  username,
  resetCreds,
  cleanAllEstimates,
  redirect,
  from,
  redirectAutomatically,
}) => (
  <Fragment>
    {redirectAutomatically && <Redirect to={from} />}
    <div>
      {/* eslint-disable-next-line react/jsx-one-expression-per-line */}
      Welcome, {username}!
    </div>
    <Button
      color="primary"
      outline
      onClick={() => redirect(from)}
    >
      Close
    </Button>
    <Button
      color="danger"
      outline
      onClick={() => {
        api.removeCreds()
        resetCreds()
        cleanAllEstimates()
      }}
    >
      Log Out
    </Button>
  </Fragment>
)

Authorized.propTypes = {
  username: PropTypes.string.isRequired,
  resetCreds: PropTypes.func.isRequired,
  cleanAllEstimates: PropTypes.func.isRequired,
  redirect: PropTypes.func.isRequired,
  from: locationType.isRequired,
  redirectAutomatically: PropTypes.bool,
}

Authorized.defaultProps = {
  redirectAutomatically: false,
}

export default Authorized
