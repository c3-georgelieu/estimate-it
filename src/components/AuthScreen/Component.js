import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import * as api from '../../helpers/api'

class AuthScreen extends React.Component {
  componentWillMount() {
    console.log('will mount')
    this.props.checkCreds()
  }

  componentDidMount() {
    const {
      match: { params },
      apiKey,
      checkCreds,
      saveCreds,
      getEstimate,
      showAuthScreen,
      // openGuestSession,
      openAuthScreen,
      closeAuthScreen,
      redirect,
      cleanEstimate,
    } = this.props
    console.log('did mount')

    if (params.estimateId !== 'new' && !apiKey) {
      cleanEstimate(params.estimateId)
      openAuthScreen()
    }

  }

  componentWillReceiveProps({
    match: { params },
    apiKey,
    checkCreds,
    saveCreds,
    getEstimate,
    openAuthScreen,
    closeAuthScreen,
    // changeText
    cleanEstimate,
  }) {
    console.log('will receive props', apiKey, 'รท',params.estimateId)

    if (params.estimateId !== 'new' && !apiKey) {
      cleanEstimate(params.estimateId)
      openAuthScreen()
    }

    // On route change
    if (params.estimateId !== this.props.match.params.estimateId) {
      if (params.estimateId === 'new') closeAuthScreen()
    }

    if (apiKey && params.estimateId) {
      if (
        apiKey !== this.props.apiKey
        ||
        params.estimateId !== this.props.match.params.estimateId
      ) getEstimate(params)
    }
  }

  render() {
    const {
      match: { params },
      apiKey,
      checkCreds,
      saveCreds,
      getEstimate,
      showAuthScreen,
      openAuthScreen,
      closeAuthScreen,
      redirect,
      openGuestSession,
    } = this.props
    console.log('render/showAuthScreen', showAuthScreen)

    return showAuthScreen && (
      <div className="auth-screen overlay">
        <form
          onSubmit={e => {
            e.preventDefault()
            api.setApiKey(e.target.apiKey.value)
            checkCreds()
          }}
        >
          <input
            type="password"
            name="apiKey"
            placeholder="Enter access key..."
            defaultValue={apiKey}
            autoFocus
          />
          <br />
          <button type="submit">
            Submit Key
          </button>
        </form>
        {apiKey ? (
          <button
            type="button"
            onClick={closeAuthScreen}
          >
            Close
          </button>
        ) : (
          <button
            type="button"
            onClick={openGuestSession}
          >
            Guest Session
          </button>
        )}
      </div>
    )
  }
}

export default AuthScreen
