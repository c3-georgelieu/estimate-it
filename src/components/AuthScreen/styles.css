.auth-screen {
  background-color: rgba(240, 248, 255, 0.5);
}

.auth-screen form label {
  display: block;
}
