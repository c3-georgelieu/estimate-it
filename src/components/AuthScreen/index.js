import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import AuthScreen from './AuthScreen'
import './styles.css'
import {
  closeAuthScreen,
  cleanEstimate,
  resetCreds,
} from '../../redux/actions'
import {
  checkCreds,
  openGuestSession,
} from '../../redux/actions/async'

const mapStateToProps = ({ creds: { apiKey } }) => ({
  apiKey,
})

const mapDispatchToProps = ({
  checkCreds,
  closeAuthScreen,
  cleanEstimate,
  openGuestSession,
  resetCreds,
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AuthScreen))
