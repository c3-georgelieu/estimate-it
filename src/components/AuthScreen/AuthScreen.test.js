import React from 'react'
import renderer from 'react-test-renderer'
import AuthScreen from './AuthScreen'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AuthScreen
        match={{ params: { estimateId: 'new' } }}
        username="Test User"
        checkCreds={_ => null}
        resetCreds={_ => null}
        cleanEstimate={_ => null}
        closeAuthScreen={_ => null}
        openGuestSession={_ => null}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
