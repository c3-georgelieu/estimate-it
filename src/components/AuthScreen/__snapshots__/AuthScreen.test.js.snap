// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="overlay auth-screen"
>
  <div
    className="panel"
  >
    <div>
      Welcome, 
      Test User
      !
    </div>
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
      type="button"
    >
      Log Out
    </button>
  </div>
</div>
`;
